arangod/VocBase/VocBaseLogManager.cpp

```diff
@@ -186,6 +184,42 @@ auto VocBaseLogManager::dropReplicatedState(arangodb::replication2::LogId id)
   return result;
 }
 
+auto VocBaseLogManager::resignAndDrop(GuardedData::StateAndLog& stateAndLog)
+    -> std::unique_ptr<replication2::storage::IStorageEngineMethods> {
+  auto& state = stateAndLog.state;
+  auto& log = stateAndLog.log;
+
+  // Get the state handle so we can drop the state later
+  auto stateHandle = log->disconnect(std::move(stateAndLog.connection));
+
+  // resign the log now, before we update the metadata to avoid races
+  // on the storage.
+  auto storage = std::move(*log).resign();
+
+  auto metadata = storage->readMetadata();
+  if (metadata.fail()) {
+    THROW_ARANGO_EXCEPTION(
+        std::move(metadata).result());  // state untouched after this
+  }
+
+  // Invalidate the snapshot in persistent storage.
+  metadata->snapshot.updateStatus(
+      replication2::replicated_state::SnapshotStatus::kInvalidated);
+  // TODO make sure other methods working on the state, probably meaning
```

Comment:
TODO still open?
I have seen this is just moved from old code

Reply:
Yes, it is from old code, was already there. It is still open, although not critical. I think we should keep it.